src/Core/Translation/Storage/Finder/TranslationFinder.php

```diff
@@ -155,4 +157,88 @@ private function getDomainFromFile(SplFileInfo $file, string $locale): string
 
         return $domain;
     }
+
+    /**
+     * Get metadata from original catalogue, parse them to extract filename and line and add them to the updated catalogue
+     *
+     * @param MessageCatalogue $catalogue
+     * @param MessageCatalogue $originalCatalogue
+     *
+     * @return MessageCatalogue
+     */
+    private function updateCatalogueMetadata(MessageCatalogue $catalogue, MessageCatalogue $originalCatalogue): MessageCatalogue
+    {
+        $locale = $originalCatalogue->getLocale();
+        $localeSuffix = '.' . $locale;
+        $suffixLength = strlen($localeSuffix);
+
+        foreach ($originalCatalogue->all() as $domain => $messages) {
+            $originalDomain = $domain;
+            // Remove locale suffix from domain name
+            if (substr($domain, -$suffixLength) === $localeSuffix) {
+                $domain = substr($domain, 0, -$suffixLength);
+            }
+            foreach (array_keys($messages) as $translationKey) {
+                $metadata = $originalCatalogue->getMetadata($translationKey, $originalDomain);
+                if ($this->shouldAddFileMetadata($metadata)) {
```

Comment:
Is it useful to add the metadata if there is no notes?

```suggestion
                if ($this->shouldAddFileMetadata($metadata) && $this-> metadataContainNotes($metadata) {
```
Wdyt